### PR TITLE
Add `seasonal_plot`

### LINKED ISSUE #548: Add seasonal plot
### 🚀 Feature Request

seasonplot1-1
Add plot function that allows to create plots similar to the above one.

### Motivation

Seasonal plot allows to analyse and extract useful info from time series.

### Proposal

```
def seasonal_plot(
    ts: "TSDataset",
    freq: Optional[str] = None
    cycle: Union[Literal["hour"], Literal["day"], Literal["week"], Literal["month"], Literal["quarter"], Literal["year"], int] = "yearly",
    align: Union[Literal["first"], Literal["last"]] = "last"
    aggregation: Union[Literal["sum'], Literal["mean"] = sum,
    in_column: str = "target",
    segments: Optional[List[str]] = None,
    columns_num: int = 2,
    figsize: Tuple[int, int] = (10, 5),
    **plot_kwargs
):
```

Where:
```
ts - the data
freq - if not None then resample data to it.
cycle - is a period of X axis
align - used only when cycle is int. Defines if last or first period should be full. 
aggregation - how to aggregate data when resampling
target - defines what column to plot
segments - allows to specify segments to plot, if None plot all the segments
columns_num - number of graphics columns
figsize - size of the figure per subplot with one segment in inches
**plot_kwargs - additional arguments that will be passed to inner plotting function
```

**Notes on cycle and freq**:
`if cycle == int and freq == None` then consider freq current and just split the data by chunks of cycle size. Use align parameter to fill first or last period.

`if cycle == int and freq != None` then resample data using freq and split the data by chunks of cycle size. Use align parameter to fill first or last period.

`if cycle == Literal and freq == None` then consider freq current and split the data using information about time:

> hour - starts at 00 minutes ends at 59 minutes
> day - starts at 00:00, ends at 23:59
> week starts at Monday 00:00 ends at Sunday 23:59
> and so on

Also consider adding custom X axis such as Weekdays[Mon, Tue, Wed, etc], Months[Jan, Feb, Mar, Apr, etc] and check if data allows using it. 

`if cycle == Literal and freq != None` then resample first using freq and split the data using information about time just as in the above case.

### Example

If data has freq = "D" then to plot weekly seasonality we need to call
`seasonal_plot(ts, cycle="week")`
it should plot a graph that will start at Monday and end on Sunday.

> Note that `seasonal_plot(ts, cycle=7)` may not start Monday because time series can start and end at any weekday possible.

If data has freq = "D" then to plot yearly seasonality as above example we need to call:
`seasonal_plot(ts, freq="M", cycle="year")`

### Test cases

Should test, that split by season works correctly

### Alternatives

_No response_

### Additional context

Examples: 
- https://github.com/giotto-ai/giotto-time/blob/9c84a8fc1b9dc5769181ce0e9e9f44976884cd41/gtime/plotting/plotting.py#L182
- https://medium.com/analytics-vidhya/automate-seasonality-plots-with-plotly-and-python-23e7fef065a2

### Checklist

- [ ] I discussed this issue with ETNA Team

### PR DESCRIPTION
**Pull request by Mr-Geekman**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Proposed Changes
Look #548.

## Related Issue
#548.

## Closing issues
Closes #548.

### CODE REVIEW

**Comment by Mr-Geekman** [comment]
Example script:
```python
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd

from etna.analysis import seasonal_plot
from etna.datasets import TSDataset


def main():
    df = pd.read_csv("examples/data/example_dataset.csv", parse_dates=["timestamp"])
    df_wide = TSDataset.to_dataset(df)
    df_wide.iloc[:10, 0] = np.NaN
    ts = TSDataset(df=df_wide, freq="D")

    # make plot
    seasonal_plot(ts=ts, freq=None, cycle="month")
    plt.savefig("seasonal_plot_day_month")

    seasonal_plot(ts=ts, freq=None, cycle="year")
    plt.savefig("seasonal_plot_day_year")

    seasonal_plot(ts=ts, freq=None, cycle="week", figsize=(10, 10))
    plt.savefig("seasonal_plot_day_week")

    seasonal_plot(ts=ts, freq="W", cycle="year", aggregation="mean")
    plt.savefig("seasonal_plot_week_year_mean")

    seasonal_plot(ts=ts, freq="M", cycle="year", aggregation="sum")
    plt.savefig("seasonal_plot_month_year_sum")

    seasonal_plot(ts=ts, freq=None, cycle=50, alignment="first")
    plt.savefig("seasonal_plot_day_50_first")

    seasonal_plot(ts=ts, freq=None, cycle=50, alignment="last")
    plt.savefig("seasonal_plot_day_50_last")


if __name__ == "__main__":
    main()
```

seasonal_plot_day_month:
seasonal_plot_day_month

seasonal_plot_day_year:
seasonal_plot_day_year

seasonal_plot_day_week:
seasonal_plot_day_week

seasonal_plot_week_year_mean:
seasonal_plot_week_year_mean

seasonal_plot_month_year_sum:
seasonal_plot_month_year_sum

seasonal_plot_day_50_first:
seasonal_plot_day_50_first

seasonal_plot_day_50_last:
seasonal_plot_day_50_last

**Comment by martins0n** [comment]
Yеt another problem with duplicated legend, isn't it?

File: etna/analysis/eda_utils.py [hunk 1]
```diff
@@ -411,3 +414,343 @@ def prediction_actual_scatter_plot(
...
+
+def _get_seasonal_cycle_name(
+    timestamp: pd.Series,
+    cycle: Union[
+        Literal["hour"], Literal["day"], Literal["week"], Literal["month"], Literal["quarter"], Literal["year"], int
+    ],
+) -> pd.Series:
+    """Get unique name for each cycle in a series with timestamps."""
+    if isinstance(cycle, int):
+        row_numbers = pd.Series(np.arange(len(timestamp.sort_values())))
```
[the remark this answers is not in the document]
Comment (Mr-Geekman) [reply etna/analysis/eda_utils.py:498]: I'll remove sorting.

File: etna/analysis/eda_utils.py [hunk 2]
```diff
@@ -411,3 +414,343 @@ def prediction_actual_scatter_plot(
...
+        return (row_numbers // cycle + 1).astype(str)
+    elif SeasonalPlotCycle(cycle) == SeasonalPlotCycle.hour:
```
[the remark this answers is not in the document]
Comment (Mr-Geekman) [reply etna/analysis/eda_utils.py:500]: Where is better to put this dict? Beginning of the function?

File: tests/test_analysis/test_eda_utils.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
+import numpy as np
```
[the remark this answers is not in the document]
Comment (Mr-Geekman) [reply tests/test_analysis/test_eda_utils.py:1]: Tests on running? Ok, I'll add them

File: etna/analysis/eda_utils.py [hunk 4]
```diff
@@ -411,3 +414,343 @@ def prediction_actual_scatter_plot(
...
+    cycles_df["in_cycle_num"] = _get_seasonal_in_cycle_num(
+        timestamp=cycles_df["timestamp"], cycle_name=cycles_df["cycle_name"], cycle=cycle, freq=freq
+    )
+    cycles_df["in_cycle_name"] = _get_seasonal_in_cycle_name(
+        timestamp=cycles_df["timestamp"], in_cycle_num=cycles_df["in_cycle_num"], cycle=cycle, freq=freq
+    )
+    return cycles_df
+
+
+def _resample(df: pd.DataFrame, freq: str, aggregation: Union[Literal["sum"], Literal["mean"]]) -> pd.DataFrame:
```
[the remark this answers is not in the document]
Comment (Mr-Geekman) [reply etna/analysis/eda_utils.py:609]: I thought that resample functions here a specific. May be we should create a separate task for adding resampling.

[the remark this answers is not in the document]
Comment (martins0n) [reply etna/analysis/eda_utils.py:609]: But it looks like ordinal resample 🤔

File: etna/analysis/eda_utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (martins0n) [reply etna/analysis/eda_utils.py:500]: Yes

**Review by martins0n — APPROVED** [review]
🥇

**Pull request merged**